services/notification_queue/notification_queue.py

```diff
@@ -1068,7 +1068,7 @@ async def post(self):
                                             session.commit()
                                             queue.append(notification.id)
                                 elif is_followup_request:
-                                    if target_data["status"] == "submitted":
+                                    if "submitted" in target_data["status"]:
```

Comment:
could be an issue. For some apis we might have a failed status that could contain "submitted" in it.

Instead, please do `target_data["status"].startswith()`